device/WakeOnLan/controller.js

```diff
@@ -0,0 +1,155 @@
+﻿'use strict';
+
+const BluePromise = require('bluebird');
+const fs = require('fs')
+const readline = require('readline')
+const os = require('os');
+const net = require('net');
+const dgram = require('dgram');
+
+const PORT_NUMBER = 9;
+const DEFAULT_IPADDRESS = '255.255.255.255';
+
+let wolMap = "wol.map";
+
+/**
+ * Sets the path to the 'wol.map' file
+ */
+module.exports.setMapPath = (mapPath) => {
+    wolMap = mapPath;
+}
+
+/**
+ * Handles button presses for WOL
+ */
+module.exports.onButtonPressed = (deviceid, name) => {
+    console.log(`WOL button pressed for ${name}`);
+
+    parseAddr(name)
+        .then((addr) => {
+            return Promise.all([addr, getMacBuffer(addr.macAddress)])
+        })
+        .then((results) => {
+            let magic = Buffer.alloc(6, 0xff);
+            for (let i = 0; i < 16; i++) {
+                magic = Buffer.concat([magic, results[1]]);
+            }
+            return Promise.all([results[0], magic]);
+        })
+        .then((results) => {
+            return sendPacket(results[0].ipAddress, results[1]);
+        })
+        .then((results) => {
+            console.log("WOL packet sent");
```

Comment:
I would use the `debug` package to output debug information, see https://github.com/visionmedia/debug. 
I would add this output before the `return sendPacket...` so you can omit the additional then branch